test/test_variable_unlim.jl

```diff
@@ -22,7 +22,7 @@ NCDatasets.NCDataset(filename,"c") do ds
 
         for j = 1:sz[2]
             data[:,j] .= T(j)
-            v[:,j] .= T(j)
+            v[:,j] = fill(T(j), sz[1])
```

Comment:
This is where we need that `resize!` write function I've been going on about.

Unlimited dimension writing needs an API so it doesn't go through the normal `setindex!` machinery, or there will be bugs like this everywhere.

(Also from my perceptive Rasters.jl integration really needs CFVariable to be a DiskArray.jl sooner or later, or it will be lazy reads or CF standards, but not both)